### PR TITLE
beeper: 3.71.16 -> 3.80.17

### PR DESCRIPTION
**Pull request by mjm**
## Description of changes

Bumping to latest version.

More importantly, I added an update script. It seems like the current version has a somewhat different filename structure than before, so I removed the `${version}` in the construction of the URL in favor of just using exactly what is provided in the `latest-linux.yml`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/beeper/default.nix
Comment (jshcmpbll) [inline pkgs/applications/networking/instant-messengers/beeper/default.nix:17]: So strange they changed this but oh well. I also tried using the direct URL mentioned [here](https://github.com/NixOS/nixpkgs/pull/241992#issuecomment-1730304937) but that doesn't seem to be working either..

Comment (mjm) [reply pkgs/applications/networking/instant-messengers/beeper/default.nix:17]: Yes, I was trying to use that as well but it wouldn't work so I gave up on that.

File: pkgs/applications/networking/instant-messengers/beeper/default.nix
Comment (jshcmpbll) [inline pkgs/applications/networking/instant-messengers/beeper/default.nix:69]: Really clean. Thanks for introducing this to me! I haven't been the best at keeping up on a lot of this so I appreciate you contributing the `updateScript` so I could learn about it. Ill have to give it a try for the other package I maintain.

**Review by jshcmpbll — APPROVED** [review]
🔥

File: pkgs/applications/networking/instant-messengers/beeper/default.nix
Comment (jshcmpbll) [reply pkgs/applications/networking/instant-messengers/beeper/default.nix:69]:
```
Going to be running update for following packages:
 - beeper-3.80.17

Press Enter key to continue...

Running update for:
 - beeper-3.80.17: UPDATING ...
 - beeper-3.80.17: DONE.

Packages updated!
```

**Comment by mjm** [comment]
Awesome! Once this merges, I also have another PR I'll open that should fix the wrapper so that this correctly passes the right flags for Wayland, which the current version doesn't do because it's using the QT mkDerivation which uses the binary wrapper instead of the shell wrapper.

**Comment by jshcmpbll** [comment]
Sounds good! 🥳  I am not on Wayland so I just took suggestions from other users when the derivation was first opened. You're welcome to join as a contributor too if you are interested.

**Comment by mjm** [comment]
You mean a maintainer? If so, I'd be happy to.

Was there a reason you used `qt5.callPackage` for Beeper? AFAICT it doesn't use QT, and even if it did, I'm not sure if the QT stuff in Nixpkgs would work correctly on an AppImage. The change I have prepared to fix the wrapper for Wayland changes it to use the basic callPackage and mkDerivation instead, but I'll have to reconsider if it was needed for something.

And one more question: do recent versions of Beeper crash often for you? I can't get it to stay open for even a minute at a time anymore, and I'm wondering if that's specific to me.

**Comment by jshcmpbll** [comment]
Ah, yes, sorry haha. Would love to have you!

Honestly probably not needed, `qt5.callPackage`. There was a lot of back and forth and I am not sure if I contributed that dependency or of someone else did but we can certainly give it a go without it.

I have good news for you! Yes, beeper was crashing constantly for me and I even reached out to support but I found the following reddit post the other day that fixed it for me! - https://www.reddit.com/r/beeper/comments/16p6oxl/linux_version_crashing/

I dont know if I should open an issue in nixpkgs or just add a comment to the derivation but I think a lot of people using beeper on nix are probably having the same problem. (not that it is specific to nix as the reddit user is saying they are on ubuntu)

**Comment by mjm** [comment]
Omg thank you, I'm trying that now, here's hoping it works for me too (it's looking good so far)! And yeah, I'm not sure what the right way is to communicate this known issue.

I'll push another commit adding myself as a maintainer :)

**Review by jshcmpbll — APPROVED** [review]

**Review by jocampbell-exelixis — APPROVED** [review]

**Review by jshcmpbll — APPROVED** [review]

**Comment by Emiller88** [comment]
Awesome @mjm! Does this automatically update when rebuilding locally? I was going to add a package version override so something like:

```
(beeper.override {version = "3.79.17";})
```
Would be possible for users.

But is that even necessary now with this fix?

**Comment by mjm** [comment]
I don't think that would be possible for a few different reasons, but mainly it comes down to not having an easy mapping from version to download URL. The update script can (a) only get the URL for the latest version and (b) requires network access to do it, which wouldn't be available during the build. And you would also need to know and provide the hash of the different version you're using.

Anyway, I don't think it will be needed going forward. We should be able to keep the version in Nixpkgs up-to-date without pain now. I think @r-ryantm will be able to open PRs for it automatically based on using Repology and the update script, but if not, it will still be pretty easy for us to create them ourselves.

**Pull request merged**
